container/openejb-core/src/main/java/org/apache/openejb/threads/impl/ContextServiceImplFactory.java

```diff
@@ -77,6 +86,36 @@ public static ContextServiceImpl newPropagateEverythingContextService() {
     }
 
 
+    public static ContextServiceImpl getOrCreateDefaultSingleton() {
+        if (defaultSingleton == null) {
+            defaultSingleton = newDefaultContextService();
+        }
+
+        return defaultSingleton;
+    }
+
+    public static ContextServiceImpl lookupOrDefault(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
thinking out loud: I guess the default case would be ok to use the default singleton without logging so maybe we can have a specific token for that ("tomeeDefault" - very bad name but it should not be something xbean layer handles (`@xxx`, `$...`) nor something too common like `default`, maybe `auto!`).
This way we can get `if (auto) return singleton else if (not set) return logAndSingleton() else lookup()` and get a nominal default case.

I'll let you judge on this one, current flavor is fine for me once it gets the lazy logger.

Reply:
I've changed the default ContextService name to `[automatic]` now for SMES/MES which will cause the factory to use the singleton. Please take a look if I've done things as you intended 😉